### DISCUSSION TITLE
Combat screen, when the player interacts with the action bar and the troops on the battlefield, indicate in gray the location of the troops on the battlefield and in the action bar [Done]

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4417] [Ideas]
When the player positions the mouse cursor over the slot of the creature troop sprite in the action bar, the hexagon(s) where the troop is located on the battlefield should be grayed out.

When the player positions the mouse cursor over the hexagon(s) where a troop is located on the battlefield, the slot of the creature troop sprite in the action bar should be grayed out.
The background color of the creature sprites should be lighter by default, so that a grayer background can be applied when needed.
If more than one troop can be affected by the area of effect of an attack or a shot, all troop slots in the action bar are grayed out.

When the player wants to cast a spell on a troop and positions the mouse cursor over the hexagon(s) where a troop is located on the battlefield, the slot of the creature troop sprite in the action bar should be grayed out.
If more than one troop can be affected by the area of effect of the spell, all troop slots in the action bar are grayed out.

When the player wants to cast a spell on a troop and positions the mouse cursor over a slot of the creature troop sprite in the action bar, the hexagon(s) where the troop is located on the battlefield should be grayed out.
Eventually, the player should also be able to cast the spell by selecting the troop in the action bar (when possible, so for spells that affect a single creature, and maybe the area effect spells if possible).
The spell cursor should be displayed on the location of the targeted troop in the action bar when it is possible to cast the spell.

In fheroes2:

![Interaction action bar battlefield flèche 001](https://user-images.githubusercontent.com/43583994/137303176-c7fbfffc-5e33-494e-a104-82dfbd168bc6.png)

![Interaction action bar battlefield flèche 002](https://user-images.githubusercontent.com/43583994/137303202-3077e2d8-e7ba-4614-9857-9d3a76afdf40.png)

![Interaction action bar battlefield flèche 003](https://user-images.githubusercontent.com/43583994/137303220-fe49227a-8f24-4b2b-90df-3b16ba476d08.png)

![Interaction action bar battlefield flèche 004](https://user-images.githubusercontent.com/43583994/137304139-af4feea2-47e9-4019-955d-ee1577252f67.png)

**Comment by Branikolog** [comment]
Hi, @LeHerosInconnu 
Nice Suggestion! :)
I think the highlighting of creatures in action bar should be made with red frame. The same it was implemented for artifacts and creatures in hero meeting dialog. It's much more distinctive and suits the UI style. What do you think?

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> Hi, @LeHerosInconnu
> Nice Suggestion! :)

Thanks. :)

> I think the highlighting of creatures in action bar should be made with red frame. The same it was implemented for artifacts and creatures in hero meeting dialog. It's much more distinctive and suits the UI style. What do you think?

I had also thought about this red frame, but a troop that has already played (like the Gargoyles troop in the example) is already in a red frame (although it doesn't flash) which then can be confusing.
And I like better the fact that there is a match between the grey hexagon and the troop box with the grey background. :)

**Reply by Branikolog** [reply]
@LeHerosInconnu 
I think, that we shouldn't stick to a current implementation and can change the color or the way game highlights icons: we can change gargoyles to a gray (as it's more calm color) (OR even darken the icon a bit) and make selected troop red, as targeting in Heroes2 is often implemented with red color.
And the gray hex itself looks not gray, actually on different terrains. :)

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> @LeHerosInconnu I think, that we shouldn't stick to a current implementation and can change the color or the way game highlights icons: we can change gargoyles to a gray (as it's more calm color) (OR even darken the icon a bit) and make selected troop red, as targeting in Heroes2 is often implemented with red color.

Or maybe animate the little creature icon in the action bar?

> And the gray hex itself looks not gray, actually on different terrains. :)

"Grey" is the way of expressing that the area of the hexagon becomes darker. A more appropriate term would be "Shadow". :D

**Comment by LeHerosInconnu** [comment]
This has been implemented.
I close the discussion.
